lib/marin/src/marin/rl/health_monitor.py

```diff
@@ -0,0 +1,591 @@
+        regressions = sum(
+            1
+            for env_id, m in self.env_metrics.items()
+            if m.current_state == "graduated"
+            and env_id in self.graduation_baselines
+            and m.current_eval_performance
+            < self.graduation_baselines[env_id]["performance"] * self.config.regression_threshold
+        )
+
+        return {
+            "total_environments": total_envs,
+            "health_distribution": {"healthy": healthy, "warning": warning, "critical": critical},
+            "state_distribution": {"locked": locked, "active": active, "graduated": graduated},
+            "graduated_regressions": regressions,
+            "active_warnings": sum(len(w) for w in self.active_warnings.values()),
+            "average_training_steps": (
+                np.mean([m.total_training_steps for m in self.env_metrics.values() if m.total_training_steps > 0])
+                if any(m.total_training_steps > 0 for m in self.env_metrics.values())
+                else 0
+            ),
+        }
+
+    def get_environment_report(self, env_id: str) -> dict[str, Any] | None:
+        """Get detailed health report for a specific environment.
+
+        Args:
+            env_id: Environment identifier.
+
+        Returns:
+            Detailed health report or None if environment not found.
+        """
+        if env_id not in self.env_metrics:
+            return None
+
+        metrics = self.env_metrics[env_id]
+        report = {
+            "env_id": env_id,
+            "current_state": metrics.current_state,
+            "health_status": metrics.health_status.value,
+            "performance": {
+                "current_train": metrics.current_train_performance,
+                "current_eval": metrics.current_eval_performance,
+                "peak_train": metrics.peak_train_performance,
+                "peak_eval": metrics.peak_eval_performance,
+                "trend": metrics.performance_trend,
+                "variance": metrics.performance_variance,
+            },
+            "training": {
+                "total_steps": metrics.total_training_steps,
+                "time_in_state": metrics.time_in_current_state,
+                "steps_since_progress": metrics.steps_since_last_progress,
+            },
+            "warnings": [{"type": w.type.value, "message": w.message, "severity": w.severity} for w in metrics.warnings],
+        }
+
+        if metrics.graduation_performance is not None:
+            report["graduation"] = {
+                "performance": metrics.graduation_performance,
+                "step": metrics.graduation_step,
+                "current_ratio": (
+                    metrics.current_eval_performance / metrics.graduation_performance
+                    if metrics.graduation_performance > 0
+                    else 0
+                ),
+            }
+
+        return report
+
+    def should_send_wandb_alert(self, warning_type: WarningType) -> bool:
```

Comment:
as mentioned, maybe pass in now=time.time() here so you can test without sleeping.